src/AdminUtils.js

```diff
@@ -67,40 +79,72 @@ class AdminUtils {
   }
 
   add(discord, message, params) {
+    if (params[0] == 'help') {
+      return message.reply('add <clip> [category] (with attachment): \n' +
+          'Adds a sound effect with <clip> as its shortcut.\n' +
+          'If provided, will assign to [category]. Defaults to "misc".');
+    }
     if (!this._paramCheck(message, params)){ return; }
 
-    const prefix = params[0];
-    if (!prefix.match(/^[a-z0-9_]+$/)) {
-      return message.reply(`${prefix} is a bad short name`);
+    const clipName = params[0];
+    const category = params[1] || 'Misc';
+    if (!clipName.match(/^[a-z0-9_]+$/)) {
+      return message.reply(`${clipName} is a bad short name`);
+    }
+    if (!category.match(/^[a-z0-9_]+$/)) {
+      return message.reply(`${category} is a bad category name`);
     }
-    if (Object.keys(files).indexOf(prefix) > -1) {
+    if (fm.inLibrary(clipName)) {
       return message.reply("That sound effect already exists");
     }
     if (message.attachments.first()) {
       // Only check the first attachment
-      const a = message.attachments.first();
-      const filename = `./Uploads/${prefix}--${a.filename}`;
-
-      log.debug(`Writing attachment to file: ${filename}`);
-      request(a.url).pipe(fs.createWriteStream(filename));
-
-      files[prefix] = filename;
-      return message.reply(`${nconf.get('KEY_SYMBOL')}${prefix} is now available`);
+      fm.create(clipName, category, message.attachments.first());
+      return message.reply(`${nconf.get('KEY_SYMBOL')}${clipName} is now available`);
     } else {
       return message.reply(`You need to attach a file`);
     }
   }
 
+  categorize(discord, message, params) {
+    if (params[0] == 'help') {
+      return message.reply('categorize `<new category>` `<clip>` [`<clip>` ...]: \n' +
+        'Updates the category for any sound(s) (space separated).\n'+
+        'Remember that categories should use `_` for spaces.');
+    }
+    if (!this._paramCheck(message, params, 2)){ return; }
+    const category = params.shift();
+    params.forEach(clip => {
+      if(fm.inLibrary(clip)) {
+        fm.rename(clip, clip, category);
+        message.reply(`${clip}'s category is now: ${category}`);
+      } else {
+        message.reply(`I don't recognize ${clip}`)
+      }
+    });
+  }
+
   grant(discord, message, params) {
-    if (!this._paramCheck(message, params)){ return; }
+    if (params[0] == 'help') {
+      return message.reply('grant `<username>` `<permission>`[,`<permission>`...]: \n' +
```

Comment:
We should try to keep the tool syntax consistent. Here we use comma separated values for parameters, but for categories we use space separated.